Move controller specs to request specs

arun1595 opened the pull request:
#66

phallstrom commented on spec/requests/membership_submissions_request_spec.rb:
```diff
@@ -0,0 +1,55 @@
+require "rails_helper"
```
I think this file is redundant with the next one? Only change is in the file name "request" ?

arun1595 replied:
Ah, you're right. This file should be in the commits. Let me remove it.

arun1595 replied:
_Update_: The redundant file has been removed.

phallstrom commented on spec/requests/membership_submissions_spec.rb:
```diff
@@ -0,0 +1,55 @@
+  before do
+    slack_link = "slack://channel?id=xxxx&team=yyyy"
+
+    allow_any_instance_of(ApplicationHelper).to receive(:link_to_open_slack)
```
Instead of this, what do you think about creating the general slack channel? Feels like that might be better and would cover later cases where we may do something similar instead of overriding this one particular helper method which doesn't really have anything to do with membership submissions?

We could even go so far as to create that general slack channel in a before block in the rails helper so it's available to all tests.

Thoughts?

arun1595 replied:
Yeah, I agree creating the "general" slack channel is a much better approach.

> We could even go so far as to create that general slack channel in a before block in the rails helper so it's available to all tests.

Sure. Do we want it created for all tests? Won't it suffice to create it only for request specs?

Let me know if I'm missing something.

phallstrom commented on spec/requests/slack_post_event_spec.rb:
```diff
@@ -0,0 +1,104 @@
+    let(:token) { "valid" }
+
+    before do
+      allow(ENV).to receive(:[]).with("SLACK_VERIFICATION_TOKEN").and_return(token)
```
This makes me a little nervous if only because it will break if I change the code from `[]` to use `fetch`.   Maybe we could pull in https://github.com/ljkbennett/stub_env or maybe we can commit a `.env.test` file that has a fake value defined and reference that?

arun1595 replied:
That's a good point. We could add in a `.env.test` with a fake value but I would like to set it explicitly in the tests which gives us the flexibility to stub different values for different specs. I am okay with using in the `stub_env` gem.

What do you think?

arun1595 commented:
@phallstrom I've updated the PR making use of `stub_env` to stub ENV variables. I've added a helper to create a general slack channel (if it doesn't exist) and configured it to run before every group of examples are run.

PTAL when you get a chance.

Thanks!
